analytical_engine/core/fragment/arrow_projected_fragment.h

```diff
@@ -399,6 +402,52 @@ class ArrowProjectedFragment
 
   ~ArrowProjectedFragment() {}
 
+  static bl::result<std::shared_ptr<vineyard::ArrowFragment<oid_t, vid_t>>>
+  MergeGraphAndContext(
+      const grape::CommSpec& comm_spec,
+      const std::shared_ptr<
+          ArrowProjectedFragment<oid_t, vid_t, vdata_t, edata_t>>&
+          projected_fragment,
+      const std::shared_ptr<IVertexDataContextWrapper>& ctx_wrapper) {
+    vineyard::ObjectMeta projected_frag_meta = projected_fragment->meta();
+    label_id_t projected_label_id =
+        projected_frag_meta.GetKeyValue<label_id_t>("projected_v_label");
+    VLOG(1) << "projected label id: " << projected_label_id;
+
+    // First get out arrow array
+    std::vector<std::pair<std::string, std::string>> selector_list;
+    selector_list.emplace_back("result", "r");
+    std::string selector_string = generate_selectors(selector_list);
+    auto selector = Selector::ParseSelectors(selector_string).value();
+    BOOST_LEAF_AUTO(arrow_arrays,
+                    ctx_wrapper->ToArrowArrays(comm_spec, selector));
```

Comment:
The `ToArrowArray` will do a copy.

The copy can be avoid, but it's fine for current implementation for simplification. Please add a `TODO` or `FIXME` here.

Reply:
We could allocate the context on vineyard directly, but its fine to keep the current codepath. Anyway add a TODO.